Formula/samba.rb

```diff
@@ -0,0 +1,54 @@
+class Samba < Formula
+  desc "SMB/CIFS file, print, and login server for UNIX"
+  homepage "https://samba.org/"
+  license "GPL-3.0-or-later"
+  url "https://download.samba.org/pub/samba/samba-4.14.5.tar.gz"
+  sha256 "bb6ef5d2f16b85288d823578abc453d9a80514c42e5a2ea2c4e3c60dc42335c3"
+
+  depends_on "pkg-config" => :build
+  depends_on "gnutls"
+  depends_on "openssl"
+  depends_on "readline" # Without the readline dependency the build fails on macOS 10.14+
+
+  # https://bugzilla.samba.org/show_bug.cgi?id=14680
+  patch do
+    url "https://attachments.samba.org/attachment.cgi?id=16579"
+    sha256 "86fce5306349d1c8f3732ca978a31065df643c8770114dc9d068b7b4dfa7d282"
+  end
+
+  def install
+    # Samba can be used to share directories with the guest in QEMU user-mode (SLIRP) networking
+    # with the `-net nic -net user,smb=/share/this/with/guest` option.
+    # The shared folder appears in the guest as "\\10.0.2.4\qemu".
+    system "./configure",
+           "--disable-cephfs",
+           "--disable-cups",
+           "--disable-iprint",
+           "--disable-glusterfs",
+           "--disable-python",
+           "--without-acl-support",
+           "--without-ad-dc",
+           "--without-ads",
+           "--without-dnsupdate",
+           "--without-ldap",
+           "--without-libarchive",
+           "--without-json",
+           "--without-ntvfs-fileserver",
+           "--without-pam",
+           "--without-regedit",
+           "--without-syslog",
+           "--without-utmp",
+           "--without-winbind",
+           "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+    # macOS has its own SMB daemon as /usr/sbin/smbd, so rename our smbd to samba-dot-org-smbd to avoid conflict.
+    # samba-dot-org-smbd is used by qemu.rb .
+    mv "#{sbin}/smbd", "#{sbin}/samba-dot-org-smbd"
+  end
+
+  test do
+    system "#{sbin}/samba-dot-org-smbd", "--version"
+    system "#{sbin}/samba-dot-org-smbd", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Added test script